src/main/java/core/basesyntax/service/impl/ReportMakerServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.dao.StorageDao;
+import core.basesyntax.db.dao.StorageDaoImpl;
+import core.basesyntax.service.ReportMakerService;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReportMakerServiceImpl implements ReportMakerService {
+    private final String delimiter;
+    private final String header;
+    private final StorageDao storageDao;
+
+    public ReportMakerServiceImpl(String delimiter, String header) {
+        this.delimiter = delimiter;
+        this.header = header;
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public List<String> makeReport() {
+        List<String> report = new ArrayList<>();
+        report.add(header);
+        for (String fruit : storageDao.getAll().keySet()) {
+            report.add(fruit + delimiter + storageDao.get(fruit));
```

Comment:
I'm not sure, but if you have a method getAll() in storageDao that gives access to the entire map, then do you even need a class StorageDao?